packages/build-tools/src/utils/outputs.ts

```diff
@@ -0,0 +1,80 @@
+import { Generic } from '@expo/eas-build-job';
+import { BuildStep, jsepEval } from '@expo/steps';
+import { bunyan } from '@expo/logger';
+import nullthrows from 'nullthrows';
+import fetch from 'node-fetch';
+
+import { BuildContext } from '../context';
+
+export async function uploadJobOutputsToWwwAsync(
+  ctx: BuildContext<Generic.Job>,
+  {
+    steps,
+    logger,
+    expoApiV2BaseUrl,
+  }: { steps: BuildStep[]; logger: bunyan; expoApiV2BaseUrl: string }
+): Promise<void> {
+  if (!ctx.job.outputs) {
+    logger.info('Job defines no outputs, skipping upload');
+    return;
+  }
+
+  try {
+    const workflowJobId = nullthrows(ctx.job.builderEnvironment?.env?.__WORKFLOW_JOB_ID);
+    const robotAccessToken = nullthrows(ctx.job.secrets?.robotAccessToken);
+
+    const interpolationContext = {
+      steps: Object.fromEntries(steps.map((step) => [step.id, getDynamicValuesFromStep(step)])),
+    };
+    logger.info({ dynamicValues: interpolationContext }, 'Using dynamic values');
+
+    const outputs = getJobOutputsFromSteps({
+      jobOutputDefinitions: ctx.job.outputs,
+      interpolationContext,
+    });
+    logger.info('Uploading outputs');
+
+    await fetch(new URL(`workflows/${workflowJobId}`, expoApiV2BaseUrl).toString(), {
+      method: 'PATCH',
+      body: JSON.stringify({ outputs }),
+      headers: {
+        Authorization: `Bearer ${robotAccessToken}`,
+        'Content-Type': 'application/json',
+      },
+      timeout: 20000,
+    });
```

Comment:
I believe that `fetch` won't throw if the status code is for example 500 by default. You need to check it manually. It will only throw on things like timeouts.
```suggestion
    const response = await fetch(new URL(`workflows/${workflowJobId}`, expoApiV2BaseUrl).toString(), {
      method: 'PATCH',
      body: JSON.stringify({ outputs }),
      headers: {
        Authorization: `Bearer ${robotAccessToken}`,
        'Content-Type': 'application/json',
      },
      timeout: 20000,
    });
   if (!response.ok) {
     throw new Error(...)
   }
```